atc/db/resource_cache_lifecycle.go

```diff
@@ -46,6 +47,14 @@ func (f *resourceCacheLifecycle) CleanUsesForFinishedBuilds(logger lager.Logger)
 	return err
 }
 
+func (f *resourceCacheLifecycle) CleanDirtyInMemoryBuildUses(lager.Logger) error {
+	_, err := sq.Delete("resource_cache_uses").
+		Where(sq.Expr("((now() - in_memory_build_create_time) > '24 HOURS'::INTERVAL)")).
```

Comment:
Maybe we should use `GlobalResourceCheckTimeout` + a short period, like we do here: https://github.com/concourse/concourse/blob/1e72106a49b406405db8e8a75770680f7d07db1e/atc/atccmd/command.go#L1213-L1220

I know it's currently buggy (#7260), but will be fixed by either #7426 or #6597